presto-orc/src/main/java/com/facebook/presto/orc/reader/LongDictionaryProvider.java

```diff
@@ -65,17 +83,36 @@ public LongDictionaryProvider(InputStreamSources dictionaryStreamSources)
      * @param items number of items expected in the dictionary.
      * @return The DictionaryResult contains two parts:
      * 1) the final dictionary buffer object. Different from the input dictionary buffer if the input
-     *    dictionary buffer is expanded or that the method returns a shared dictionary.
+     * dictionary buffer is expanded or that the method returns a shared dictionary.
      * 2) whether the caller will be the owner of the dictionary, for the purpose of memory accounting.
-     *    Callers own all non-shared dictionaries, and only the first caller of the shared dictionary
-     *    is the owner.
+     * Callers own all non-shared dictionaries, and only the first caller of the shared dictionary
+     * is the owner.
      * @throws IOException
      */
     public DictionaryResult getDictionary(StreamDescriptor streamDescriptor, long[] dictionary, int items)
             throws IOException
     {
-        InputStreamSource<LongInputStream> dictionaryDataStream = dictionaryStreamSources.getInputStreamSource(streamDescriptor, DICTIONARY_DATA, LongInputStream.class);
-        return loadDictionary(streamDescriptor, requireNonNull(dictionaryDataStream), dictionary, items);
+        InputStreamSource<LongInputStream> dictionaryDataStream = requireNonNull(dictionaryStreamSources.getInputStreamSource(streamDescriptor, DICTIONARY_DATA, LongInputStream.class));
```

Comment:
why do we have requireNonNull here ? From your previous explanation, we always receive MissingStreamSource here.